EssentialsGeoIP/src/com/earth2me/essentials/geoip/EssentialsGeoIPPlayerListener.java

```diff
@@ -183,9 +189,9 @@ private void downloadDatabase() {
             InputStream input = conn.getInputStream();
             OutputStream output = new FileOutputStream(databaseFile);
             byte[] buffer = new byte[2048];
-            if (url.endsWith(".gz")) {
+            if (url.contains("gz")) {
```

Comment:
This feels clowny. endsWith makes it better but obviously is a generally shitty solution. Could we possibly pick up the GZIP magic numbers and make decisions off of that?

Reply:
We could but I don't know there's much of a point in doing so. I just changed it for now because the format of the URL changed. It used to be a direct link to a file with the `.tar.gz` suffix, but now it is a query parameter without the preceding `.` which may appear anywhere in the link. Feel free to suggest any code you think might work more cleanly.

Reply:
fuck it i guess? its a lot more code.